broker/src/main/java/io/zeebe/broker/system/partitions/ZeebePartition.java

```diff
@@ -382,4 +383,42 @@ public void triggerSnapshot() {
   public ActorFuture<Optional<StreamProcessor>> getStreamProcessor() {
     return actor.call(() -> Optional.ofNullable(context.getStreamProcessor()));
   }
+
+  public ActorFuture<Optional<ExporterDirector>> getExporterDirector() {
+    return actor.call(() -> Optional.ofNullable(context.getExporterDirector()));
+  }
+
+  public ActorFuture<Void> pauseExporting() {
+    final CompletableActorFuture<Void> completed = new CompletableActorFuture<>();
+    actor.call(
+        () -> {
+          try {
+            context.pauseExporting();
+
+            if (context.getExporterDirector() != null && !context.shouldExport()) {
+              context.getExporterDirector().pauseExporting().onComplete(completed);
+            } else {
+              completed.complete(null);
+            }
+          } catch (final IOException e) {
+            LOG.error("Could not pause exporting", e);
+            completed.completeExceptionally(e);
+          }
```

Comment:
Not sure whether I get this. 
We first pause the exporting (which actually internally just persist the file), then we check whether there is a exporter and whether we should not export (which should always be the case i guess?!). 

Maybe it makes sense to rename the method so it is more clear what actually happens. We persist the pausing state and remove the check. I see no value in it?

Reply:
The check is added as a safeguard incase persisting of the state was not successful. Do you think that is unnecessary?

Reply:
So you actually want to check whether the persisting succeeded so maybe you check the return value of the method which is called persistPauseState

Reply:
Then it is I think more clear. Do you agree?